addons/website_sale/models/sale_order.py

```diff
@@ -419,7 +437,49 @@ def _notify_get_recipients_groups(self, msg_vals=None):
         return groups
 
     def action_confirm(self):
-        res = super(SaleOrder, self).action_confirm()
+        for order in self:
+            order_location = order.access_point_address
+
+            if not order_location:
+                continue
+
+            # retrieve all the data :
+            # name, street, city, state, zip, country
+            name = order.partner_shipping_id.name
+            street = order_location['pick_up_point_address']
+            city = order_location['pick_up_point_town']
+            zip_code = order_location['pick_up_point_postal_code']
+            country = order.env['res.country'].search([('code', '=', order_location['pick_up_point_country'])]).id
+            state = order.env['res.country.state'].search(['&', ('code', '=', order_location['pick_up_point_state']), ('country_id.id', '=', country)]).id if (order_location['pick_up_point_state'] and country) else None
+            parent_id = order.partner_shipping_id.id
+            email = order.partner_shipping_id.email
+            phone = order.partner_shipping_id.phone
+
+            # we can check if the current partner has a partner of type "delivery" that has the same address
+            existing_partner = order.env['res.partner'].search(['&', '&', '&', '&',
+                                                                ('street', '=', street),
+                                                                ('city', '=', city),
+                                                                ('state_id', '=', state),
+                                                                ('country_id', '=', country),
+                                                                ('type', '=', 'delivery')], limit=1)
+
+            if existing_partner:
+                order.partner_shipping_id = existing_partner
+            else:
+                # if not, we create that res.partner
+                order.partner_shipping_id = order.env['res.partner'].create({
+                    'parent_id': parent_id,
+                    'type': 'delivery',
+                    'name': name,
+                    'street': street,
+                    'city': city,
+                    'state_id': state,
+                    'zip': zip_code,
+                    'country_id': country,
+                    'email': email,
+                    'phone': phone
+                })
```

Comment:
This comes from `website_sale_delivery`'s override of `_action_confirm` (which is called in `sale` at the end of `action_confirm`, for additional mechanisms on sale order confirmation), but instead, you now add it in the beginning of `action_confirm`, before even calling super... :thinking: This means you'll make all this BEFORE checking `_get_forbidden_state_confirm`, add missing subscribers that need to be aware of some changes, etc
Even if what's done shouldn't break anything (to my knowledge), that's not something we want to change now.

Reply:
I don't change anything, it is exactly how it is done in master. I can call the super before, looks like a good idea, but am I supposed to find & solve any possible bug in the lines that I move?

Reply:
No, you didn't change what's inside the override. But this code, before, overrides `_action_confirms`, and now it overrides `action_confirm`. These are not the same method.

Reply:
Oh, well spotted :pray: